pkg/operation/care/checker.go

```diff
@@ -521,9 +521,10 @@ func (b *HealthChecker) CheckLoggingControlPlane(
 // CheckExtensionCondition checks whether the conditions provided by extensions are healthy.
 func (b *HealthChecker) CheckExtensionCondition(condition gardencorev1beta1.Condition, extensionsConditions []ExtensionCondition) *gardencorev1beta1.Condition {
 	for _, cond := range extensionsConditions {
-		// check if the health check condition.lastUpdateTime is older than the configured staleExtensionHealthCheckThreshold
-		if b.staleExtensionHealthCheckThreshold != nil && Now().UTC().Sub(cond.Condition.LastUpdateTime.UTC()) > b.staleExtensionHealthCheckThreshold.Duration {
-			c := gardencorev1beta1helper.UpdatedCondition(condition, gardencorev1beta1.ConditionUnknown, fmt.Sprintf("%sOutdatedHealthCheckReport", cond.ExtensionType), fmt.Sprintf("%s extension (%s/%s) reports an outdated health status (last updated: %s ago at %s).", cond.ExtensionType, cond.ExtensionNamespace, cond.ExtensionName, time.Now().UTC().Sub(cond.Condition.LastUpdateTime.UTC()).Round(time.Minute).String(), cond.Condition.LastUpdateTime.UTC().Round(time.Minute).String()))
+		lastHeartbeatTime := *cond.Condition.LastHeartbeatTime
+		// check if the health check condition.lastHeartbeatTime is older than the configured staleExtensionHealthCheckThreshold
+		if b.staleExtensionHealthCheckThreshold != nil && Now().UTC().Sub(lastHeartbeatTime.UTC()) > b.staleExtensionHealthCheckThreshold.Duration {
```

Comment:
This is not backwards compatible since some extensions might not have revendored or released this changes while Gardener was already updated. In such situation, gardenlet would look for `lastHeartbeatTime`, however, some extensions still only update `lastUpdateTime` as before. For such resources an outdated health check will be reported now.

Please adapt as following: If `lastHeartbeatTime` is not `nil` then consider it and only it. Otherwise, fall back to `lastUpdateTime`.